src/App.jsx

```diff
@@ -1,17 +1,59 @@
+import { useState, useEffect } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
+import LikeCount from './components/LikeCount';
+import HeaderTitle from './components/HeaderTitle';
+import data from './data/messages.json';
 
 const App = () => {
+  const [messages, setMessages] = useState(data);
+  const [participants, setParticipants] = useState([]);
+
+  useEffect(() => {
+    const countParticipants = () => {
+      const participants = messages.map(message => {
+        return message.sender;
+      });
+      setParticipants([...new Set(participants)]);
+    };
+
+    countParticipants();
+  }, [messages]);
+
+
+  const updateLikes = (messageId) => {
+    setMessages(updateMessages =>
+      updateMessages.map(message => {
+        if (message.id === messageId) {
+          return { ...message, liked: !message.liked };
+        }
+        return message;
+      })
+    );
+  };
+
+  const countLikes = () => {
+    const likes = messages.reduce((totalLikes, message) => {
+      if(message.liked){
+        return totalLikes + 1;
+      }
+      return totalLikes;
```

Comment:
Nice job using `reduce` here. 

You could use a ternary here so the logic can all be in one like:
```suggestion
      return message.liked ? totalLikes + 1 : totalLikes;
```